tofhir-engine/src/main/scala/io/tofhir/engine/execution/ErroneousRecordWriter.scala

```diff
@@ -0,0 +1,78 @@
+package io.tofhir.engine.execution
+
+import io.tofhir.engine.model.{FhirMappingErrorCodes, FhirMappingJobExecution, FhirMappingResult}
+import io.tofhir.engine.util.FileUtils.FileExtensions
+import org.apache.hadoop.fs.FileUtil
+import org.apache.spark.sql.functions.{col, from_json, schema_of_json}
+import org.apache.spark.sql.{Dataset, SaveMode, SparkSession}
+
+import java.io.File
+import java.util
+
+/**
+ * Handler for post mapping processing. Spark is used to process the data sources.
```

Comment:
I think the comments need to be adjusted w.r.t. the new class names and structure for archiver and erroneous record writer.